File: CBuilder/GZipDeflate/Unit1.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "Unit1.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "DemoBaseFormUnit"
#pragma link "clGZip"
#pragma resource "*.dfm"
TForm1 *Form1;
//---------------------------------------------------------------------------
__fastcall TForm1::TForm1(TComponent* Owner)
  : TclDemoBaseForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TForm1::btnCompressClick(TObject *Sender)
{
  clGZip1->Compress(edtSource->Text, edtDestination->Text);
  ShowMessage(Format("The file %s was deflated and stored to %s", ARRAYOFCONST((edtSource->Text, edtDestination->Text))));
}
//---------------------------------------------------------------------------

void __fastcall TForm1::btnUncompressClick(TObject *Sender)
{
  TStream *source = new TFileStream(edtSource->Text, fmOpenRead);
  TStream *dest = new TFileStream(edtDestination->Text, fmCreate);
  __try {
    clGZip1->Uncompress(source, dest);
  }
  __finally {
    delete dest;
    delete source;
  }
  ShowMessage(Format("The file %s was inflated and stored to %s", ARRAYOFCONST((edtSource->Text, edtDestination->Text))));
}
//---------------------------------------------------------------------------

void __fastcall TForm1::btnSourceClick(TObject *Sender)
{
  if (OpenDialog1->Execute())
    edtSource->Text = OpenDialog1->FileName;
}
//---------------------------------------------------------------------------

void __fastcall TForm1::btnDestinationClick(TObject *Sender)
{
  if (SaveDialog1->Execute())
    edtDestination->Text = SaveDialog1->FileName;
}
//---------------------------------------------------------------------------


void __fastcall TForm1::clGZip1Progress(TObject *Sender,
      __int64 ABytesProceed, __int64 ATotalBytes)
{
  ProgressBar1->Max = ATotalBytes;
  ProgressBar1->Position = ABytesProceed;
}
//---------------------------------------------------------------------------

File: Delphi/GZipDeflate/Unit1.pas
unit Unit1;

interface

uses
  Windows, Messages, SysUtils, Variants, Classes, Graphics, Controls, Forms,
  Dialogs, StdCtrls, clGZip, ComCtrls, DemoBaseFormUnit, ExtCtrls;

type
  TForm1 = class(TclDemoBaseForm)
    btnCompress: TButton;
    btnUncompress: TButton;
    Label1: TLabel;
    Label2: TLabel;
    Label3: TLabel;
    Label4: TLabel;
    edtSource: TEdit;
    btnSource: TButton;
    edtDestination: TEdit;
    btnDestination: TButton;
    OpenDialog1: TOpenDialog;
    SaveDialog1: TSaveDialog;
    clGZip1: TclGZip;
    ProgressBar1: TProgressBar;
    procedure btnSourceClick(Sender: TObject);
    procedure btnDestinationClick(Sender: TObject);
    procedure btnCompressClick(Sender: TObject);
    procedure btnUncompressClick(Sender: TObject);
    procedure clGZip1Progress(Sender: TObject; ABytesProceed, ATotalBytes: Int64);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  Form1: TForm1;

implementation

{$R *.dfm}

procedure TForm1.btnSourceClick(Sender: TObject);
begin
  if OpenDialog1.Execute() then
  begin
    edtSource.Text := OpenDialog1.FileName;
  end;
end;

procedure TForm1.btnDestinationClick(Sender: TObject);
begin
  if SaveDialog1.Execute() then
  begin
    edtDestination.Text := SaveDialog1.FileName;
  end;
end;

procedure TForm1.btnCompressClick(Sender: TObject);
begin
  clGZip1.Compress(edtSource.Text, edtDestination.Text);
  ShowMessage(Format('The file %s was deflated and stored to %s', [edtSource.Text, edtDestination.Text]));
end;

procedure TForm1.btnUncompressClick(Sender: TObject);
var
  source, dest: TStream;
begin
  source := TFileStream.Create(edtSource.Text, fmOpenRead);
  dest := TFileStream.Create(edtDestination.Text, fmCreate);
  try
    clGZip1.Uncompress(source, dest);
  finally
    dest.Free();
    source.Free();
  end;
  ShowMessage(Format('The file %s was inflated and stored to %s', [edtSource.Text, edtDestination.Text]));
end;

procedure TForm1.clGZip1Progress(Sender: TObject; ABytesProceed, ATotalBytes: Int64);
begin
  ProgressBar1.Max := ATotalBytes;
  ProgressBar1.Position := ABytesProceed;
end;

end.
